cpp/CMake/Findglog.cmake

```diff
@@ -1,37 +1,81 @@
-# Copyright (c) Facebook, Inc. and its affiliates.
-# - Try to find Glog
-# Once done, this will define
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
 #
-# GLOG_FOUND - system has Glog
-# GLOG_INCLUDE_DIRS - the Glog include directories
-# GLOG_LIBRARIES - link these to use Glog
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set(GLUTEN_GLOG_MINIMUM_VERSION 0.4.0)
+set(GLUTEN_GLOG_VERSION 0.6.0)
+
+if(NOT BUILD_GLOG)
```

Comment:
Curious could we just find first, then check `glog_FOUND`? seems `BUILD_GLOG` can be removed.

Reply:
It's a bit tricky here. We cannot make sure user has both glog and gflags installed. If glog is found but gflags is missing, it fails. The option is for ignoring system glog then directly build glog and gflags from source. https://github.com/oap-project/gluten/pull/3638/files#diff-9df362ebf18ceb320fa6423ff9140fd2bd75906d57d92ca018febfc3a93864a9R66

Reply:
Thanks, I'm fine with it.